### DISCUSSION TITLE
Welcome to imagej_macros_and_scripts Discussions!

### DISCUSSION THREAD

**Discussion by volker-baecker** [MontpellierRessourcesImagerie/imagej_macros_and_scripts#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. 
  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by skanda58** [comment]
Hi Volker,
I am currently having issues to use the MRI wound healing tool with my macbook (MacOS Big Sur 11.2.3.)
ImageJ otherwise works fine, but the tool does not recognise the empty space between the monolayer.
I was wondering if you heard of any compatibility issues.
Thanks for your help,
Sam

**Reply by skanda58** [reply]
or could you maybe provide me a sample image, so I can check, if maybe the input file is the issue and not the tool itself?

**Comment by volker-baecker** [comment]
Hi @skanda58,
there are sometimes problems with the settings of FIJI/ImageJ. Please check the tool's [help-page](https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/wiki/Wound-Healing-Tool) (also available from the toolset by clicking on the button with the image). Make sure that the option Black background is set in dialog accessible via the menu Process>Binary>options.... Also make sure that "Area" is among the activated measurements. 

You can also find two example images on the tool's [help-page](https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/wiki/Wound-Healing-Tool) .

If you should still have problems, please send me an image and I will have a look.

Best,
Volker

**Reply by skanda58** [reply]
Hi Volker,
I used the example .tif as an input and it still does not work.
Black background is already set.
Also I made sure to crop the specific area with (tissue - gap - tissue) to reduce analysis time.
I made screenshots with the example image - I will attach it to this message. I also added my DMSO control wound healing assay wells.

Thanks for your support!

Best,
Sam


![SK_MA1_220114_Cal27_MRT_0h_A3_confluence_b686ed1d-ae11-4209-a924-7fd4a74c204d](https://user-images.githubusercontent.com/80246855/149994611-ab8bc6fb-3813-4949-8cc3-e74f5cbb14e3.jpg)

![SK_MA1_220114_Cal27_MRT_16h_A3_confluence_58b697c1-d922-4c15-89fd-d349b7a4d95e](https://user-images.githubusercontent.com/80246855/149994561-b523ad55-2282-4f84-b7cd-87178c4b3c24.jpg)
<img width="1434" alt="Bildschirmfoto 2022-01-18 um 19 08 22" src="https://user-images.githubusercontent.com/80246855/149994755-90df4960-ce6c-40f5-a3ac-f45cbee073f4.png">
<img width="1425" alt="Bildschirmfoto 2022-01-18 um 19 08 10" src="https://user-images.githubusercontent.com/80246855/149994773-264e7f29-a1d7-419e-9a42-07c093fd97a7.png">
<img width="1400" alt="Bildschirmfoto 2022-01-18 um 19 08 02" src="https://user-images.githubusercontent.com/80246855/149994778-942a07f7-e256-4211-a8c8-625a8ffae8c1.png">

**Reply by skanda58** [reply]
I just tried to analyze my input files on a windows PC with ImageJ and the mri wound healing tool and still did not work out.
Looking forward to hearing from you,
Sam

**Reply by volker-baecker** [reply]
Hi @skanda58,
did you use the parameters "variance, 20, 1, 4 and 999999" for the example image as suggested on the [help-page](https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/wiki/Wound-Healing-Tool) ?

I see that you opened the image as virtual stack, although this works it is not helpful for the tool, since the image is duplicated before processing anyway and therefore completely loaded into memory.

Your own images do not look like images for the tool. For the tool apart from the gap there should be tissue in the image everywhere. Still by cropping and using the right parameters it could work. If you can provide an original input image (not a screenshot) I can try.
Best,
Volker

**Comment by Svethna** [comment]
Hello, I am interested in your tool to identify layer and measure their thickness which is SkinTool. I managed to use your plugin for one channel and measure one layer but it does not work for others. Is there any possibility to use it for my analysis ? ( I attached an image to show) Thank you for your help !
![81 VP merge CaBP VGlut2 x40 lob6-7](https://user-images.githubusercontent.com/116018641/196226412-3c1409f6-0455-4da7-8dc1-6dd08ab70c53.jpg)

**Comment by volker-baecker** [comment]
Hi @Svethna,
is that the complete image or just a region? You run the tool on a binary mask, so you have to create that first. The tool will (sometimes) not work correctly if the mask touches the upper or lower border of the image. Now the red channel you cannot really analyze, at least not at the left and right parts, because we cannot see the lower border and cannot know the thickness there.  Depending on how you segment the green channel, it might touch the upper border, however only minimally, we did not really loose the thickness information, so you can do on the mask of the grreen channel: select all, cut, paste, move it down some pixels. Now it won't touch the border any more and the tool will work.

![thickness](https://user-images.githubusercontent.com/7602420/196437876-9a50feff-a9cc-4b8f-b09c-813ca24adea3.png)

Should your problem be related to something else, please send me the raw input images, so that I can test.

Please also note that it only measures the "thickness" in some sense. The 'm'-button measures vertically across the object from local extrema of the lower border, while the 't'-button measures across the object, perpendicular to the lower border, at random positions.

Please let me now if you'd prefer different measurements, I'd be happy to add them.
Best,
Volker

**Comment by Svethna** [comment]
Hello, Thank you very much for your fast reply, it is very appreciated. I posted only the region I am interested in measuring parameters. I use images acquired with a microscope with 2 fluorescence channels that I merge with ImageJ. Until now, to use the plugin I would then switch to RGB stack to make a threshold on each channel where each color marks a layer. In my mind, the threshold gives an upper and lower limit of the layer of interest. Then I wanted to make consecutive measurements parallel to each other but perpendicular to the surface, all along the layer. I found that for my green channel I could more or less achieve what I wanted (although I had to resize the image and flip it so that the layer was roughly horizontal, otherwise the measurements were no longer perpendicular to the surface) but I couldn't get a result for the red. I don't know if I'm clear enough, I'm attaching some raw images as well as an image showing my region of interest and I hope you can help me.I really thank you in advance for your time. Please find the files on this wetransfer link : https://we.tl/t-wWkt9BgjXD Have a nice day.Regards, Svethna  De : VolkerEnvoyé le :mardi 18 octobre 2022 15:11À : MontpellierRessourcesImagerie/imagej_macros_and_scriptsCc : Svethna; MentionObjet :Re: [MontpellierRessourcesImagerie/imagej_macros_and_scripts] Welcome to imagej_macros_and_scripts Discussions! (Discussion #5) Hi @Svethna,is that the complete image or just a region? You run the tool on a binary mask, so you have to create that first. The tool will (sometimes) not work correctly if the mask touches the upper or lower border of the image. Now the red channel you cannot really analyze, at least not at the left and right parts, because we cannot see the lower border and cannot know the thickness there. Depending on how you segment the green channel, it might touch the upper border, however only minimally, we did not really loose the thickness information, so you can do on the mask of the grreen channel: select all, cut, paste, move it down some pixels. Now it won't touch the border any more and the tool will work.Should your problem be related to something else, please send me the raw input images, so that I can test.Please also note that it only measures the "thickness" in some sense. The 'm'-button measures vertically across the object from local extrema of the lower border, while the 't'-button measures across the object, perpendicular to the lower border, at random positions.Please let me now if you'd prefer different measurements, I'd be happy to add them.Best,Volker—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Reply by volker-baecker** [reply]
Hi  @Svethna,
why wouldn't you simply use FIJI's local-thickness plugin for that? Make a mask of your region of interest, run the plugin and you will get a table with the thickness values (calculated from the distance map, i.e. perpendicular to the centerline). Does that solve your problem?
Best,
Volker

**Comment by Svethna** [comment]
Hi again, As far as I understood, it applies only for 3D images. The skintool had the best result so far that is why I wanted to find a way to apply it to all the layers of my tissue. Regards,Svethna   De : VolkerEnvoyé le :mardi 18 octobre 2022 17:03À : MontpellierRessourcesImagerie/imagej_macros_and_scriptsCc : Svethna; MentionObjet :Re: [MontpellierRessourcesImagerie/imagej_macros_and_scripts] Welcome to imagej_macros_and_scripts Discussions! (Discussion #5) Hi @Svethna,why wouldn't you simply use FIJI's local-thickness plugin for that? Make a mask of your region of interest, run the plugin and you will get a table with the thickness values (calculated from the distance map, i.e. perpendicular to the centerline). Does that solve your problem?Best,Volker—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by volker-baecker** [comment]
Hi @Svethna,
no, the local thickness also works with 2D images. I just tried it on your image and the result looks fine to me.
Please try it. I'm sure it is much closer to what you need than the skin-tool.
Best,
Volker

**Comment by Svethna** [comment]
Hello again, I might not be using the Local Thickness plugin the way you did. I end up with this image (after changing type into 8 bit then thresholding to create a mask) giving a visual representation of local thickness I guess. The given value are shown in the table and the histogram but do not correspond of what I expect of layer thickness… Thank you once again for your help. Regards,Svethna  De : VolkerEnvoyé le :mardi 18 octobre 2022 17:41À : MontpellierRessourcesImagerie/imagej_macros_and_scriptsCc : Svethna; MentionObjet :Re: [MontpellierRessourcesImagerie/imagej_macros_and_scripts] Welcome to imagej_macros_and_scripts Discussions! (Discussion #5) Hi @Svethna,no, the local thickness also works with 2D images. I just tried it on your image and the result looks fine to me.Please try it. I'm sure it is much closer to what you need than the skin-tool.Best,Volker—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by volker-baecker** [comment]
Hi @Svethna 
please look here [Width-Profile-Tools](https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/wiki/Width-Profile-Tools) for different possibilities to measure width-profiles. I would recommend the Voronoi-method.
Best,
Volker

**Reply by Svethna** [reply]
Hello, what you did is wonderfull. I tested it but the same error is coming, whether I test it on my images or even on the dataset included in the plugin. I have the latest version with all the plugins that are required installed. I can successfully use the "local thickness" tool or the voronoi but when i want to use "perpendicular to inertia axis" this error codes comes up : 
Error:		Unrecognized Image function in line 54:
		(called from line 16)
	
		Image . <removeScale> ( ) ; 

Can you please help me to solve it ? 

Thanks once again for your kind help.
Best,
Svethna

**Reply by volker-baecker** [reply]
Hi @Svethna,
the command ``Image.removeScale()`` has been recently added to ImageJ. You need at least ImageJ 1.53u, in which the command has been introduced. It should work with the latest version 1.54d. So, please update your ImageJ (Help>Update ImageJ...) and try again. 

If there should be anymore problems, you encounter,  I would aks you to create an Issue, rather than reporting it here.

Best,
Volker

**Comment by Clairedf** [comment]
Hi,
I would like to use Analyze Spheroid Cell Invasion In 3D Matrix for my invasion assay.
I would like to obtain the area of invasion (as on the exemple shown).
But I have this error.

![Screenshot 2022-11-30 175921](https://user-images.githubusercontent.com/119524207/204860966-3c2116be-d791-40d2-8ee3-f47950316ee8.png)

What should I do?
Best,
Claire

**Reply by volker-baecker** [reply]
Hi @Clairedf,
please install MorphoLibJ. See [Analyze-Spheroid-Cell-Invasion-In-3D-Matrix](https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/wiki/Analyze-Spheroid-Cell-Invasion-In-3D-Matrix) for full instructions.
Best,
Volker

**Reply by Clairedf** [reply]
Hi,
thank you for your answer!
It's work now. I tried for a couple of spheroid I have (from different cell lines), and I have some issue to not recognize the background as "invasion". (pictures below). I that case, the background is recognize as invasive area even if in the "reverse contrast" it's easy to visualise by eye the edge.
Do  you have any tips of parameters setup to obtain a nice spheroid area? 
Best,
Claire 
![Screenshot 2022-12-01 100321](https://user-images.githubusercontent.com/119524207/205265391-0ec93785-7866-4946-8049-266d5654fa96.png)

**Comment by chaipotter** [comment]
Hi,
I'm using the MRI Wound Healing Tool and having trouble with batch processing.
For each image in the folder, the tool give the results in one .xls file. If I have 50 images to process, I end up with 50 .xls files, so I have to open all of them, copy/paste 1 line of each file to another Excel file for further statistical analyzes.
Is there a way to combine results from batch processing into just one file?
Thank you very much.

**Reply by volker-baecker** [reply]
Hi @chaipotter,,

usually we apply the Wound Healing Tool on a time-series (in the form of a stack, i.e. one image with many frames). In that case each xls-files will contain one line per frame. It seems that you have images with just one frame, why is that? Is that the individual frames of a time-series or are that independent images? In the first case, you could import them all as a stack into ImageJ and run the Wound Healing Tool on that. 

Best,
Volker

**Comment by isabelmelo16** [comment]
Hi, I'm trying to use the MRI Opera export tools. I read a 96 well plate (48 wells) in the Phenix for alexa 488. I exported the raw tiff files and need to stitch them. However when I installed the plug ins, I'm encountering a IndexError: index out of range: 17
How can I approach this?

**Reply by volker-baecker** [reply]
Hi @isabelmelo16,
sorry for the late reply, I've been on holiday. I can not tell, without more information. Could you send the complete output from the log. The best would be if I could have your dataset to test, if that is possible. 
Best,
Volker

**Comment by isabelmelo16** [comment]
Hi Mr. Volker,

The files are 3.5 GB, please let me know if you have trouble opening them.
These are exports from the OPERA PHENIX. The images are named
row_column_field_channel
I imaged 2 channels.

Your help is greatly appreciated,

Isabel Melo
 Images.zip
<https://drive.google.com/file/d/1wlYFP2jIK4TMu-pMF-j_BivtygUe0cNJ/view?usp=drive_web>


El lun, 15 jul 2024 a las 5:57, Volker ***@***.***>)
escribió:

> Hi @isabelmelo16 <https://github.com/isabelmelo16>,
> sorry for the late reply, I've been on holiday. I can not tell, without
> more information. Could you send the complete output from the log. The best
> would be if I could have your dataset to test, if that is possible.
> Best,
> Volker
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/discussions/5#discussioncomment-10049293>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2FOTC76ENQRY54U746MMSLZMOMHBAVCNFSM6AAAAABKTJSXKGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMBUHEZDSMY>
> .
> You are receiving this because you were mentioned.Message ID:
> <MontpellierRessourcesImagerie/imagej_macros_and_scripts/repo-discussions/5/comments/10049293
> @github.com>
>


-- 

*Isabel Melo-Escobar, MS*
Phone: +1 (561)709-9283 | https://orcid.org/0000-0003-3176-9181 | LinkedIn
<https://www.linkedin.com/in/isabel-melo-escobar-ab917b149>

**Reply by volker-baecker** [reply]
Hi @isabelmelo16,

our tool was made for the harmony (the software used by the opera) 5 format. You are using the harmony 6 format. Between 5 and 6 the structure of the index file has changed. Mostly data that was uselessly repeated on each image entry has been moved to a more global level.  Since the fields in your index file (v6) do not correspond to what the tool, written for v5, expects, you get the error.

We actually have a version of the tool that handles both formats and is also faster. That version is not publicly available right now.  If you are interested please contact me via email (see my github profile).

Best,
Volker

**Comment by thoma671** [comment]
Hi,

I am trying to use the MRI Opera Export Tools but am not having much luck with selecting the index file. I have imported the raw data straight from Harmony and the index file is currently in a .xml file. The index file is able to open normally on ImageJ (without using the export tool files).

I am getting the Index File not found! error. The Harmony version is 5.1.

I have checked the file path and there are no special characters. I have reinstalled the export files in the toolsets folder but it is still not working. 

Any help is much appreciated! 

Thanks :)

**Reply by volker-baecker** [reply]
Hi @thoma671,
for the 5.1 it should work. Can you send the complete path to the index file on your system? The file is supposed to have the name "Index.idx.xml" and normally you just have to select the folder containing that file.
What is the error message you get? 
Best,
Volker

**Reply by thoma671** [reply]
Hi Volker,

Here is the complete path: "C:\Users\madel\OneDrive\23_24_Masters\2024_07_23 mCherry_resistance_assay MT019\Vemurafenib_Plate1_W1__20240802T11_14_13_Measurement1\Images\Index.xml"

I am getting the error "Index File not found !". 

The index file only has the .xml extension. I downloaded it straight from Harmony but I may have missed a step to make sure it was in the correct format? However, Harmony indicated that it only saves the index file as .xml. 

Thanks

**Reply by volker-baecker** [reply]
Hi @thoma671,

Please rename the index file to "Index.idx.xml" and try again. The opera export tools expect  it to have this name. (Please be careful to have windows show the file extension (an option somewhere), so that you do not mistake Index.idx.xml for Index.idx.xml.xml).

Best,
Volker

**Reply by thoma671** [reply]
Hi Volker,

Thank you! It worked! 

I am now having trouble locating the files after exporting them. I am able to see the options file in the parental folder but no extra folders. 

This is from the Log on imageJ. 
 
Argument parser Loaded
Arguments parsed
Traceback (most recent call last):
  File "<string>", line 2303, in <module>
  File "<string>", line 39, in main
  File "<string>", line 402, in __init__
  File "<string>", line 410, in configureFromOptions
  File "<string>", line 690, in getWells
  File "<string>", line 2225, in getWell
  File "<string>", line 705, in __init__
  File "<string>", line 2260, in getImage
ValueError: invalid literal for int() with base 10: '2024-08-02T11:15:22.4338004+12:00'
 
The eagle has landed!!!

Thanks

**Reply by volker-baecker** [reply]
Hi @thoma671,

can you send me the index file? It looks as if it's not what the software expects. As you can see from the error message it gets a timestamp where it expects an integer. 
The results will be written to sub-folders of the parent folder of the images. This is done so that the user does not have to open the image folder, which is frequently impossible, because it contains too many images.
I'm also surprised that you could open the index file with imagej. It has to be a rather small dataset.

Best,
Volker

**Reply by thoma671** [reply]
Hi Volker,

Here is the index file. 

https://1drv.ms/u/s!Akr9HWpDHlobhIBVSEiLe9cGXLh6hQ?e=KRCMxM

Thanks :)

**Reply by volker-baecker** [reply]
Hi @thoma671,
as you can see it says "HarmonyV6" in the header. 
So I can only repeat what I said before:
We actually have a version of the tool that handles both formats and is also faster. That version is not publicly available right now. If you are interested please contact me via email (see [here](https://www.mri.cnrs.fr/fr/annuaire.html)).
Best,
Volker

**Reply by juliomateoslangerak** [reply]
Hi @thoma671,

I don't know your use case but bio-formats tools (OMERO too) could be an option:
https://forum.image.sc/t/reading-data-into-fiji-from-a-perkinelmer-harmony-archive/84306/15

J

**Reply by volker-baecker** [reply]
Hi @thoma671 and @isabelmelo16 ,
I just made a version of the [opera export tools]( https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/wiki/MRI_Opera_export_tools) available that handles both formats, harmony 5 and harmony 6, and is also faster than its predecessor.
